src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpUseImplicitTypeDiagnosticAnalyzer.cs

```diff
@@ -151,6 +154,57 @@ protected override bool TryAnalyzeVariableDeclaration(TypeSyntax typeName, Seman
             return false;
         }
 
+        private bool TryAnalyzeDeclarationExpression(
+            DeclarationExpressionSyntax declarationExpression,
+            SemanticModel semanticModel,
+            OptionSet optionSet,
+            CancellationToken cancellationToken)
+        {
+            // It's not always safe to convert a decl expression like "Method(out int i)" to 
+            // "Method(out var i)".  Changing to 'var' may cause overload resolution errors.  
+            // Have to see if using 'var' means not resolving to the same type as before.  
+            // Note: this is fairly expensive, so we try to avoid this if we can by seeing if
+            // there are multiple candidates with the original call.  If not, then we don't
+            // have to do anything.
+            if (declarationExpression.Parent is ArgumentSyntax argument &&
+                argument.Parent is ArgumentListSyntax argumentList &&
+                argumentList.Parent is InvocationExpressionSyntax invocationExpression)
+            {
+                var memberGroup = semanticModel.GetMemberGroup(invocationExpression.Expression, cancellationToken);
+                if (memberGroup.Length == 1)
+                {
+                    // Only one member found.  Don't do the expensive check, this change 
+                    // should be ok.
```

Comment:
❓ Why 'should be' instead of 'is'?

❓ Does this produce an incorrect result currently if the user is calling a generic method, where the `out T` variable is actually used for type inference?

Reply:
will improve hte phrasing.

Reply:
investigating htat case now.

Reply:
> Does this produce an incorrect result currently if the user is calling a generic method, where the out T variable is actually used for type inference?

Yes.  It looks like we get that wrong all the time.  Will fix that as well.

Reply:
💭 It might be possible to simply exclude this fast path when the method is generic. As I mentioned in my other comment, the performance tool will be the guide for figuring out the specific cases with maximum performance-improving impact, but it's not ready to use yet.

Reply:
Ok.  We now check in the generic case as well.

Reply:
> It might be possible to simply exclude this fast path when the method is generic.

Yes.  That's the approach i ended up taking.  :)